File: frontend/src/lib/models/asset.model.ts
import type { AssetStatus } from "../enums/asset-status.enum";

export interface IAsset {
  id: string;
  /**
   * Creation timestamp
   */
  createdAt: string;
  /**
   * Number of images
   */
  imageCount: number;
  /**
   * Product name
   */
  name: string;
  /**
   * Product price
   */
  price: number;
  /**
   * Product URL
   */
  productUrl: string;
  status: AssetStatus;
}

File: backend/internal/database/models/asset.model.go
package models

import "github.com/gianghp123/Vidmerce/backend/internal/core/enums"

type AssetEntity struct {
	BaseItem
	// Creation timestamp
	CreatedAt string `dynamodbav:"createdAt"`
	// Number of images
	ImageCount int `dynamodbav:"imageCount"`
	// Product name
	Name string `dynamodbav:"name"`
	// Product price
	Price float64 `dynamodbav:"price"`
	// Product URL
	ProductURL string            `dynamodbav:"productUrl"`
	Status     enums.AssetStatus `dynamodbav:"status"`
}
